src/ol/renderer/canvas/VectorTileLayer.js

```diff
@@ -72,18 +72,22 @@ class CanvasVectorTileLayerRenderer extends CanvasTileLayerRenderer {
      */
     this.overlayContext_ = createCanvasContext2D();
 
-    const overlayCanvas = this.overlayContext_.canvas;
-    overlayCanvas.style.position = 'absolute';
-    overlayCanvas.style.transformOrigin = 'top left';
-
-    const container = document.createElement('div');
-    const style = container.style;
-    style.position = 'absolute';
-    style.width = '100%';
-    style.height = '100%';
+    let container = null;
 
-    container.appendChild(baseCanvas);
-    container.appendChild(overlayCanvas);
+    const overlayCanvas = this.overlayContext_.canvas;
+    if (overlayCanvas.style) {
```

Comment:
When your `createCanvas` fallback has a `style`, you don't need this condition. The only change here that makes sense would be adding a `createContainer(baseCanvas, overlayCanvas)` method, which you can override in your custom renderer.
```js
  createContainer(baseCanvas, overlayCanvas) {
    const container = document.createElement('div');
    const style = container.style;
    style.position = 'absolute';
    style.width = '100%';
    style.height = '100%';

    container.appendChild(baseCanvas);
    container.appendChild(overlayCanvas);

    return container;
  }
```